### ISSUE TITLE
restore fails with file does not exists as .gz, but it exists with .pgbi

### ISSUE THREAD

**Issue by neiaberau** [pgbackrest/pgbackrest#2130]
using gBackRest 2.46, doing some tests with file bundling and block level backup I run into this error on restore: 

restore command used
`pgbackrest --stanza=pg_main --log-level-console=info --delta --pg1-path=/var/lib/postgresql/13/main_restore restore`

```
2023-07-30 08:00:11.982 P00   INFO: restore command begin 2.46: --buffer-size=16777216 --delta --exec-id=602454-88d6225f --log-level-console=info --log-level-file=detail --pg1-path=/var/lib/postgresql/13/main_restore --process-max=2 --repo1-path=/var/lib/postgresql/backup/blklvl --stanza=pg_main
2023-07-30 08:00:12.242 P00   INFO: repo1: restore backup set 20230716-171505F_20230729-230002I, recovery will start at 2023-07-29 23:00:02
2023-07-30 08:00:12.261 P00   INFO: remap data directory to '/var/lib/postgresql/13/main_restore'
2023-07-30 08:00:12.265 P00 DETAIL: check '/var/lib/postgresql/13/main_restore' exists
2023-07-30 08:00:12.265 P00 DETAIL: remove 'global/pg_control' so cluster will not start if restore does not complete
.....
.....
2023-07-30 08:17:37.901 P00  ERROR: [055]: raised from local-2 protocol: unable to open missing file '/var/lib/postgresql/backup/blklvl/backup/pg_main/20230716-171505F/pg_data/pg_log/postgresql-2023-06-29_215213.log.gz' for read
```
  

this file does not exist, but it does exist with the ending pgbi

```
 find /var/lib/postgresql/backup/blklvl/backup/pg_main  -name "postgresql-2023-06-29_215213.log*"
/var/lib/postgresql/backup/blklvl/backup/pg_main/20230716-171505F/pg_data/pg_log/postgresql-2023-06-29_215213.log.pgbi
```

_now the question is, what did I do wrong or is it and error?_

However, if do 2 restores, instead of directly to the newest incremental, first the full then the current incremental, then it works:
```
pgbackrest --stanza=pg_main --log-level-console=info --delta --pg1-path=/var/lib/postgresql/13/main_restore --set 20230716-171505F restore
#followed be 2nd restore
pgbackrest --stanza=pg_main --log-level-console=info --delta --pg1-path=/var/lib/postgresql/13/main_restore  restore
#no errrors
```

_details_
pgBackRest 2.46,

pgbackrest.conf
```
[global]
log-level-file=detail

process-max=2
start-fast=y
buffer-size=16777216


repo1-path=/var/lib/postgresql/backup/blklvl
repo1-retention-full=1
repo1-retention-diff=1
repo1-bundle=y
repo1-block=y
repo1-retention-archive-type=diff

[pg_main]
pg1-path=/var/lib/postgresql/13/main
```

backup commands used:
```
/usr/bin/pgbackrest --stanza=pg_main --type=full backup
/usr/bin/pgbackrest --stanza=pg_main --type=diff backup
/usr/bin/pgbackrest --stanza=pg_main --type=incr backup

```

**Comment by dwsteele** [comment]
This should be fixed by #2123. Can you retest with 2.47?

This may be hard to reproduce, though, because it depends on a file reaching a certain age where it would not use block incremental. What you'll need to do is keep your original full backup if you still have it and create a new diff off that full backup. With 2.47 the diff should work correctly (and with 2.46 it won't).

> However, if do 2 restores, instead of directly to the newest incremental, first the full then the current incremental, then it works

This works because the file is correctly stored in the full manifest and does not change in the incremental, so delta restore does not need the info in the incremental manifest.

**Comment by neiaberau** [comment]
will do the retesing with 2.47, will close this for now
